src/test/java/test/dataprovider/issue2504/IssueTest.java

```diff
@@ -0,0 +1,25 @@
+package test.dataprovider.issue2504;
+
+import java.util.Collections;
+import org.assertj.core.api.Assertions;
+import org.testng.TestNG;
+import org.testng.annotations.Test;
+import org.testng.xml.XmlClass;
+import org.testng.xml.XmlTest;
+import test.SimpleBaseTest;
+
+public class IssueTest extends SimpleBaseTest {
+
+  @Test(description = "GITHUB-2504")
+  public void ensureParametersCopiedOnConfigFailures() {
+    XmlTest xmltest = createXmlTest("2504_suite", "2504_test");
+    xmltest.setVerbose(2);
+    xmltest.setXmlClasses(Collections.singletonList(new XmlClass(SampleTestCase.class)));
+    TestNG testNG = create(Collections.singletonList(xmltest.getSuite()));
+    SampleTestCaseListener listener = new SampleTestCaseListener();
+    testNG.addListener(listener);
+    testNG.run();
+    Assertions.assertThat(listener.getParameters()).hasSize(1);
```

Comment:
map the list and then assert it contains only `1, 2, 3, 4, 5`

Reply:
Fixed